demes/demes.py

```diff
@@ -388,10 +389,11 @@ class DemeGraph:
     :ivar description: A human readable description of the demography.
     :ivar time_units: The units of time used for the demography. This is
         commonly ``years`` or ``generations``, but can be any string.
-    :ivar generation_time: The generation time of demes.
-        TODO: The units of generation_time are undefined if
-        ``time_units="generations"``, so we likely need an additional
-        ``generation_time_units`` attribute.
+        This field is intended to be useful for documenting a demography,
```

Comment:
It took me a few minutes to see that the units cancel out here, and it all works once your time values are all the same units (including generation_time). We should probably document this a bit more clearly.